lib/eunit/src/eunit_data.erl

```diff
@@ -47,73 +43,83 @@
+         | {string(), tests()}
+         | {generator, fun (() -> tests())}
+         | {generator, M::moduleName(), F::functionName()}
+         | {spawn, tests()}
+         | {spawn, Node::atom(), tests()}
+         | {timeout, Time::number(), tests()}
+         | {inorder, tests()}
+         | {inparallel, tests()}
+         | {inparallel, N::integer(), tests()}
+         | {with, X::any(), [abstract_testfunction()]}
+         | {setup, proc(),
+                   setup_fun(),
+                   cleanup_fun(),
+                   tests() | instantiator()
+           }
+         | {setup, setup_fun(), cleanup_fun(), tests() | instantiator()}
+         | {setup, proc(), setup_fun(), tests() | instantiator()}
+         | {setup, setup_fun(), tests() | instantiator()}
+         | {foreach, proc(),
+                     setup_fun(),
+                     cleanup_fun(),
+                     [tests() | instantiator()]
+           }
+         | {foreach, setup_fun(), cleanup_fun(), [tests() | instantiator()]}
+         | {foreach, proc(), setup_fun(), [tests() | instantiator()]}
+         | {foreach, setup_fun(), [tests() | instantiator()]}
+         | {foreachx, proc(),
+                      setup_x_fun(),
+                      cleanup_x_fun(),
+                      Pairs::[{X::any(), abstract_instantiator()}]
+           }
+         | {foreachx, setup_x_fun(), cleanup_x_fun(),
+                      Pairs::[{X::any(), abstract_instantiator()}]}
+         | {foreachx, proc(), setup_x_fun(),
+                      Pairs::[{X::any(), abstract_instantiator()}]}
+         | {foreachx, setup_x_fun(),
+                      Pairs::[{X::any(), abstract_instantiator()}]}
+         | {node, Node::atom(), tests() | instantiator()}
+         | {node, Node::atom(), Args::string(), tests() | instantiator()}.
+-type simple_test() :: testfunction() | {Line::integer(), simple_test()}.
+-type testfunction() :: fun (() -> any())
+      | {test, M::moduleName(), F::functionName()}
+      | {M::moduleName(), F::functionName()}.
+-type abstract_testfunction() :: fun ((X::any()) -> any()).
+-type instantiator() :: fun ((R::any()) -> tests())
+                      | {with, [abstract_testfunction()]}.
+-type setup_fun() :: fun (() -> (SetupInfo::any())).
+-type cleanup_fun() :: fun ((SetupInfo::any()) -> any()).
+-type setup_x_fun() :: fun ((X::any()) -> SetupInfo::any()).
+-type cleanup_x_fun() :: fun ((X::any(), SetupInfo::any()) -> any()).
+-type abstract_instantiator() :: fun ((X::any(), SetupInfo::any()) -> tests()).
+-type proc() :: local | spawn | {spawn, Node::atom()}.
+
 %% Note that `{string(), ...}' is a short-hand for `{string(), {...}}'
 %% if the tuple contains more than two elements.
-%%
-%% @type moduleName() = atom()
-%% @type functionName() = atom()
-%% @type appName() = atom()
-%% @type fileName() = string()
+
+-type moduleName() :: atom().
+-type functionName() :: atom().
+-type appName() :: atom().
+-type fileName() :: string().
```

Comment:
Almost everywhere in this module types are written in snake case, but not those ones, we could align those to snake case as well, if you don't mind.